ext/drx/drx.c

```diff
@@ -2520,6 +2520,22 @@ drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
  * as well as AVX2 gather.
  */
 
+typedef struct _drx_state_machine_params_t {
+    byte *pc;
+    byte *prev_pc;
+    int detect_state;
+    byte *restore_dest_mask_start;
+    byte *restore_scratch_mask_start;
+    int skip_unknown_instr_count;
+    reg_id_t the_scratch_xmm;
+    reg_id_t gpr_bit_mask;
+    reg_id_t gpr_save_scratch_mask;
+    uint scalar_mask_update_no;
```

Comment:
How about brief comments for some of these: not sure what this one is.

Reply:
Comments added, done.